app/frontend/wallet/shelley/shelley-ledger-crypto-provider.ts

```diff
@@ -51,21 +53,39 @@ const ShelleyLedgerCryptoProvider = async ({network, config, isWebUSB}) => {
 
   const version = await ledger.getVersion()
 
-  checkVersion()
+  checkVersion('MINIMAL')
 
   const isHwWallet = () => true
   const getWalletName = () => 'Ledger'
 
-  const deriveXpub = CachedDeriveXpubFactory(derivationScheme, async (absDerivationPath) => {
-    const response = await ledger.getExtendedPublicKey(absDerivationPath)
-    const xpubHex = response.publicKeyHex + response.chainCodeHex
-    return Buffer.from(xpubHex, 'hex')
-  })
+  const exportPublicKeys = async (derivationPaths) => {
+    if (hasRequiredVersion(version, 'BULK_EXPORT')) {
+      return await ledger.getExtendedPublicKeys(derivationPaths)
+    }
+    const response = []
+    for (const path of derivationPaths) {
+      response.push(await ledger.getExtendedPublicKey(path))
+    }
+    return response
+  }
 
-  function checkVersion(recommended: boolean = false) {
-    if (!hasMinimalVersion(version, recommended)) {
-      const errorName = recommended ? 'NotRecommendedCardanoAppVerion' : 'OutdatedCardanoAppError'
-      throw NamedError(errorName, {
+  const {deriveXpub, cleanXpubCache} = CachedDeriveXpubFactory(
+    derivationScheme,
+    config.shouldExportPubKeyBulk,
+    async (derivationPaths) => {
+      const response = await exportPublicKeys(derivationPaths)
+      return response.map((res) => Buffer.from(res.publicKeyHex + res.chainCodeHex, 'hex'))
+    }
+  )
+
+  function checkVersion(requiredVersionType: string) {
```

Comment:
`requiredVersionType` is kind of cryptic, my first hunch was whether this was about minor/major/patch and only afterwards I realized this is about the feature. Something like `ensureFeatureSupported(featureName)` would imho make for a more intuitive interface

Reply:
Thanks :+1:, fixed